core/src/property/mod.rs

```diff
@@ -2,8 +2,80 @@ pub mod backend;
 pub mod traits;
 pub mod value;
 
+use std::fmt::Display;
+
 pub use backend::Backends;
 pub use traits::{FromActiveType, FromEntity, InitializeWith, PropertyError};
 pub use value::YrsString;
 
+use serde::{Deserialize, Serialize};
+
 pub type PropertyName = String;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum PropertyValue {
+    // Numbers
+    I16(Option<i16>),
+    I32(Option<i32>),
```

Comment:
It strikes me as odd that the Options are internal here. Wouldn't it be more sensible to use Option<PropertyValue> rather than this?